pkg/storaged/content-views.jsx

```diff
@@ -961,12 +1118,43 @@
                            }
                        }),
         ],
+        update: (dlg, vals, trigger) => {
+            if (vals.purpose == 'block' && (trigger == "layout" || trigger == "pvs" || trigger == "purpose")) {
+                for (const lay of layouts) {
+                    lay.disabled = (vals.pvs.length < lay.min_pvs);
+                    if (lay.value == vals.layout) {
+                        dlg.set_options("pvs", {
+                            min_selected: lay.min_pvs,
+                            explanation: min_pvs_explanation(vals.pvs, lay.min_pvs)
+                        });
+                    }
+                }
+                dlg.set_options("layout",
+                                {
+                                    choices: layouts,
+                                    explanation: layout_descriptions[vals.layout]
+                                });
+                const max = max_size(vals);
+                const old_max = dlg.get_options("size").max;
+                if (vals.size > max || vals.size == old_max)
+                    dlg.set_values({ size: max });
+                dlg.set_options("size", { max });
+            } else if (trigger == "purpose") {
+                dlg.set_options("size", { max: vgroup.FreeSize });
+            }
+        },
         Action: {
             Title: _("Create"),
             action: (vals, progress) => {
-                if (vals.purpose == "block")
-                    return vgroup.CreatePlainVolume(vals.name, vals.size, { });
-                else if (vals.purpose == "pool")
+                if (vals.purpose == "block") {
+                    if (!can_do_layouts)
+                        return vgroup.CreatePlainVolume(vals.name, vals.size, { });
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17226-20230918-111530-a92123e3-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)